### PR TITLE
F# code questions/commentary

### PR DESCRIPTION
**Pull request by StachuDotNet**
Further commentary. I'm now done with `fsharp-backend/src` and next will continue with `fsharp-backend/tests` before moving onto the client.

I skipped over Tablecloth, and there are certainly many things I don't fully understand. In any case, reviewing like this has been very helpful to me.

### CODE REVIEW

File: fsharp-backend/src/LibBackend/EventQueue.fs [hunk 1]
```diff
@@ -155,18 +155,32 @@ module WorkerStates =
...
+/// both at the "worker level" (ready to be picked up by workers),
+/// and at the "canvas level" (ready to be enqueued _for_ a worker).
+/// </summary>
+///
+/// <remarks>
+/// `host` is optional because we have it when we dequeue, but not enqueue.
+/// </remarks>
+///
+/// WHATISTHIS why do we have both 'name' and 'host'?
+/// is it anticipation of a queue existing in one canvas, but being used in another?
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibBackend/EventQueue.fs:169]: Pending question

File: fsharp-backend/src/LibBackend/EventQueue.fs [hunk 2]
```diff
@@ -155,18 +155,32 @@ module WorkerStates =
...
+  // esp. useful if we switch to a "CanvasName of string" type here, rather than just string.
+  // then this could be `(managedBy: CanvasName option)`
   (host : Option<string>)
+
   (canvasID : CanvasID)
-  (space : string)
-  (name : string)
-  (modifier : string)
+  (space : string) // WHATISTHIS? I'll add it to the /// comments.
+  (queueName : string) // or "eventName"?
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibBackend/EventQueue.fs:180]: would this be better as eventName?

File: fsharp-backend/src/LibBackend/Pusher.fs [hunk 3]
```diff
@@ -40,7 +44,7 @@ let push
   : unit =
   FireAndForget.fireAndForgetTask executionID $"pusher: {eventName}" (fun () ->
     task {
-      // TODO: handle messages over 10k
+      // TODO: handle messages over 10k (10k what? 10k bytes in size? does it fail after 10k (count) messages?)
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibBackend/Pusher.fs:47]: Pending question

File: fsharp-backend/src/LibBackend/TraceInputs.fs [hunk 4]
```diff
@@ -124,42 +130,6 @@ let listEvents (limit : Limit) (canvasID : CanvasID) : Task<List<EventRecord>> =
...
-//     ^ timestamp_constraint
-//   in
-//   Db.fetch sql ~name:"list_events" ~params:[Db.Uuid canvas_id]
-//   |> List.map ~f:(function
-//          | [module_; path; modifier] ->
-//              (module_, path, modifier)
-//          | out ->
-//              Exception.internal "Bad DB format for stored_events")
-
-
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibBackend/TraceInputs.fs:162]: Not sure why I thought it was OK to delete this. Reverting.

File: fsharp-backend/tests/TestUtils/TestUtils.fs [hunk 5]
```diff
@@ -237,7 +237,7 @@ let testUserType
 
 
 
-let hop (h : PT.Handler.T) = PT.SetHandler(h.tlid, h.pos, h)
```
Comment (StachuDotNet) [inline fsharp-backend/tests/TestUtils/TestUtils.fs:240]: :rabbit:

File: fsharp-backend/tests/Tests/Traces.Tests.fs [hunk 6]
```diff
@@ -26,7 +26,7 @@ let testTraceIDsOfTlidsMatch : Test =
   test "traceIDs from tlids are as expected" {
     Expect.equal
       "e170d0d5-14de-530e-8dd0-a445aee7ca81"
-      (Traces.traceIDofTLID 325970458UL |> string)
+      (Traces.traceIDofTLID 325970458UL |> string) // Why is this the expected value? confused.
```
Comment (StachuDotNet) [inline fsharp-backend/tests/Tests/Traces.Tests.fs:29]: Pending question

File: fsharp-backend/src/LibBackend/EventQueue.fs [hunk 7]
```diff
@@ -155,18 +155,32 @@ module WorkerStates =
...
+/// </remarks>
+///
+/// WHATISTHIS why do we have both 'name' and 'host'?
+/// is it anticipation of a queue existing in one canvas, but being used in another?
 let logQueueSize
   (queue_action : queue_action)
+
+  // better name ideas: anchorCanvas, or managedBy, or ownedBy.
+  // esp. useful if we switch to a "CanvasName of string" type here, rather than just string.
+  // then this could be `(managedBy: CanvasName option)`
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibBackend/EventQueue.fs:175]: Do any of these names appeal to you? Just trying to get away from confusing 'host'

File: fsharp-backend/src/LibBackend/TraceInputs.fs (hunk 4 above)
Comment (StachuDotNet) [reply fsharp-backend/src/LibBackend/TraceInputs.fs]: reverted

File: fsharp-backend/src/LibBackend/EventQueue.fs (hunk 7 above)
Comment (pbiggar) [reply fsharp-backend/src/LibBackend/EventQueue.fs:175]: "Host" is confusing. I have generally switched it over to `canvasName` and would like to get rid of `host` entirely, so "canvasName"?

File: fsharp-backend/src/LibBackend/EventQueue.fs [hunk 8]
```diff
@@ -155,18 +155,32 @@ module WorkerStates =
...
   (host : Option<string>)
+
   (canvasID : CanvasID)
   (space : string)
-  (name : string)
+  (queueName : string)
```
Comment (pbiggar) [inline fsharp-backend/src/LibBackend/EventQueue.fs:180]: This name isn't used anywhere else. I don't hate it but we should probably keep it as "name". An alternative could be to switch this to a PT.Handler.Spec.Cron?

File: fsharp-backend/src/LibBackend/Pusher.fs (hunk 3 above)
Comment (pbiggar) [reply fsharp-backend/src/LibBackend/Pusher.fs:47]: 10k bytes in size. I think it fails if it's too big.

File: fsharp-backend/src/LibBackend/EventQueue.fs (hunk 1 above)
Comment (pbiggar) [reply fsharp-backend/src/LibBackend/EventQueue.fs:169]: I think `host` should be called `canvasName`, but `name` is the name of the worker (as in `emit_v1 "workerName"`).

File: fsharp-backend/src/LibBackend/TraceInputs.fs [hunk 9]
```diff
@@ -27,7 +27,12 @@ module AT = LibExecution.AnalysisTypes
 module RT = LibExecution.RuntimeTypes
 
 
-// space, path, modifier
+/// (space/module, path, modifier)
+///
+/// "Space" is "HTTP", "WORKER", "REPL", etc.
+///
+/// "Modifier" options differ based on space.
+/// e.g. HTTP handler may have "GET" modifier.
```
Comment (pbiggar) [inline fsharp-backend/src/LibBackend/TraceInputs.fs:35]: Worth noting that Handlers which don't have modifiers (eg repl, worker) nearly always (but not actually always) have `_` as their modifier.

File: fsharp-backend/src/LibBackend/UserDB.fs [hunk 10]
```diff
@@ -1,7 +1,17 @@
...
+/// Supports anything relating to Datastores on a user canvas.
+/// Namely, this file is responsible for managing this user data, including CRUD and type checking.
+/// </summary>
+///
+/// <remarks>
+/// User data stores live within the `user_data` table, which is broken down by:
+/// - Dark version
+/// - canvas (by ID)
+/// - table/store ID
+/// - user version (version of the DB - incremented by any structural change to the store)
```
Comment (pbiggar) [inline fsharp-backend/src/LibBackend/UserDB.fs:11]: This isn't quite right. User version is for when the users change the type being stored. We also have a dark version tag which is for if we decide to store the data differently.

File: fsharp-backend/tests/Tests/Account.Tests.fs [hunk 11]
```diff
@@ -57,6 +57,12 @@ let testUsernameValidationWorks =
       "myusername09", Ok "myusername09"
       "paul", Ok "paul" ]
 
+// Note: the term "banned" here doesn't refer to users who have
```
Comment (pbiggar) [inline fsharp-backend/tests/Tests/Account.Tests.fs:60]: I think `reserved` would be a better name.

File: fsharp-backend/tests/Tests/Traces.Tests.fs (hunk 6 above)
Comment (pbiggar) [reply fsharp-backend/tests/Tests/Traces.Tests.fs:29]: This is just some input which corresponds to the output for this function. That is, the function takes a tlid and creates a UUID out of it using some standard UUID function (I think it's called "v5"), and I didn't want to test it with `0` so I made some number and this is the output we expect.

**Review by pbiggar — CHANGES_REQUESTED** [review]
Great improvements, thanks!

File: fsharp-backend/src/LibBackend/EventQueue.fs (hunk 7 above)
Comment (StachuDotNet) [reply fsharp-backend/src/LibBackend/EventQueue.fs:175]: canvasName is good by!

**Review by pbiggar — CHANGES_REQUESTED** [review]
Great. Good to go after typo fix.

File: fsharp-backend/src/LibBackend/TraceInputs.fs [hunk 12]
```diff
@@ -33,6 +33,9 @@ module RT = LibExecution.RuntimeTypes
 ///
 /// "Modifier" options differ based on space.
 /// e.g. HTTP handler may have "GET" modifier.
+///
+/// Handlers which don't have modifiers (e.g. repl, worker) nearly
+/// allways (but not actually alwasys) have `_` as their modifier.
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibBackend/TraceInputs.fs:38]:
```suggestion
/// always (but not actually always) have `_` as their modifier.
```

**Comment by pbiggar** [comment]
FYI, merge conflict.

**Pull request merged**